core/src/main/java/org/apache/calcite/plan/SubstitutionVisitor.java

```diff
@@ -1931,6 +1931,22 @@ public static MutableRel unifyAggregates(MutableAggregate query,
       final List<AggregateCall> aggregateCalls = new ArrayList<>();
       for (AggregateCall aggregateCall : query.aggCalls) {
         if (aggregateCall.isDistinct()) {
+          if (aggregateCall.getArgList().size() == 1) {
+            int aggIndex = aggregateCall.getArgList().get(0);
+            if (target.groupSet.asSet().contains(aggIndex)) {
```

Comment:
Could you please clarify why the size of the arguments has to be 1?

Reply:
The aggregate function containing the distinct keyword has only one parameter. Check the parameters here.

Reply:
Thanks,update the code.